python/graphstorm/gconstruct/transform.py

```diff
@@ -164,15 +164,21 @@ def pre_process(self, feats):
             # It will load all data into main memory.
             feats = feats.to_numpy()
 
+        feats[feats == None] = "MAGIC_NONE"
         if self._separator is None:
-            return {self.feat_name: np.unique(feats)}
+            category_keys = np.unique(feats)
         else:
             assert feats.dtype.type is np.str_, \
                     "We can only convert strings to multiple categorical values with separaters."
             vals = []
             for feat in feats:
                 vals.extend(feat.split(self._separator))
-            return {self.feat_name: np.unique(vals)}
+            category_keys = np.unique(vals)
+
+        category_keys = set(category_keys.flatten())
```

Comment:
you don't need to convert it into a set. you only do test once.

Reply:
I want to remove the key MAGIC_NONE from category_keys.